ext/websocket/02_websocketstream.js

```diff
@@ -370,7 +385,13 @@
         this[_earlyClose] = true;
       } else if (this[_closed].state === "pending") {
         PromisePrototypeCatch(
-          core.opAsync("op_ws_close", this[_rid], code, closeInfo.reason),
+          PromisePrototypeThen(
+            core.opAsync("op_ws_close", this[_rid], code, closeInfo.reason),
+            () => {
+              this[_closeSent] = new Date().getTime();
+              this[_pull]();
```

Comment:
this needs to be aware of `controller` as well, I'll refactor if the general direction makes sense.